src/ui-test/extension.all.test.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import * as extensionTest from './extension.test';
+import * as fs from 'fs';
+import * as marketplaceTest from './marketplace.test';
+import * as path from 'path';
+import * as webserver from '../test/app_soap';
+import { expect } from 'chai';
+import { DefaultWait, Project } from 'vscode-uitests-tooling';
+import { projectPath } from './package_data';
+import { VSBrowser } from 'vscode-extension-tester';
+
+describe('All tests', function () {
+	marketplaceTest.test();
+
+	describe('Extension tests', function() {
+		this.timeout(4000);
+		let browser: VSBrowser;
+		let workspace: Project;
+
+		before('Setup environment', async function() {
+			this.timeout(32000);
+			browser = VSBrowser.instance;
+			workspace = await prepareWorkspace(browser);
+			webserver.startWebService();
+		});
+
+		after('Clear environment', async function() {
+			await clearWorkspace(workspace);
+			webserver.stopWebService();
+		});
+
+		for (const f of walk(path.join(projectPath, 'src/ui-test/test-data'))) {
+			assert(f.endsWith('.json'), `${f} is not json file`);
+			const fileContent = fs.readFileSync(f, { encoding: 'utf8' });
+			extensionTest.test(JSON.parse(fileContent));
+		}
+	});
+
+});
+
+/**
+ * Iterates over all files which are children of `dir`.
+ * @param dir starting directory
+ * @returns iterable object of file absolute paths
+ */
+function* walk(dir: string): Iterable<string> {
+	const stack = [dir];
+
+	while (stack.length > 0) {
+		const file = stack.pop();
+		const stat = fs.statSync(file);
+
+		if (stat && stat.isDirectory()) {
+			// add directories and files to stack and transform filenames to absolute paths
+			stack.push(...fs.readdirSync(file).map(f => path.join(file, f)));
+		} else {
+			yield file;
+		}
+	}
+}
+
+/**
+ * Creates new project and opens it in vscode
+ */
+async function prepareWorkspace(browser: VSBrowser): Promise<Project> {
+	const project = new Project(extensionTest.WORKSPACE_PATH);
+	
+	expect(project.exists).to.be.false;
+	project.create();
+	expect(project.exists).to.be.true;
+	
+	await project.open();
+	await DefaultWait.sleep(12000);
```

Comment:
I could try wait for title bar to show folder name. I am going to try it.

Reply:
I would recommend to let it here for now to be able to move on and fix it with another PR, after issue [FUSETOOLS2-317](https://issues.redhat.com/browse/FUSETOOLS2-317) will be resolved. Because it is affecting more components.. WDYT? @apupier @mlorinc

Reply:
In my point of view, we should use sleep for now. New wait condition partially works but it occasionally fails.

Reply:
ok